### PR TITLE
Merge frontend2 into master

### LINKED ISSUE #33: Tag all controllers in WeekController with @RequestMapping annotations
Tag all controllers in WeekController with @RequestMapping annotations appropriately, in line with the instructions in Evan's video.

Decide whether a method corresponds to a @PutMapping, @GetMapping, or @PostMapping, and tag appropriately. Tag the beginning of the class with @RequestMapping("someUrlName") so that all methods in that class can be called with, for example, @GetMapping("localhost:8080/someUrlName/methodCall").

### LINKED ISSUE #38: Create methods for JS to build a week following successful sign-up
- Got to branch frontend-2, and create a branch from there called frontend-2-timetableJS
- Create a file in front-end/timetable/js called buildWeek.js.
- Make a POST request to 'http://localhost:8080/build-week' with the user's id and startDate
- Make a GET request to 'http://localhost:8080/weeks/{userid}' 
- Make a JS function converting this week to timetable HTML

### LINKED ISSUE #45: Application not starting due to server port problem
Application not starting because Intellij runs an application on whichever port we set our application to run on at runtime. Find out how to fix.

### PR DESCRIPTION
**Pull request by issamarabi**

### CODE REVIEW

**Review by JMH-0402 — APPROVED** [review]

**Comment by thelfenbaum** [comment]
finally! after so much work!

**Review by RunyuXia — APPROVED** [review]

**Review by thelfenbaum — COMMENTED** [review]
:)

**Comment by issamarabi** [comment]
@thelfenbaum Why did you close without asking me?????????? smh

**Review by laviealon — APPROVED** [review]

**Review by laviealon — APPROVED** [review]

File: src/main/front-end/signin/signin.css [hunk 1]
```diff
@@ -76,10 +76,7 @@ form.form1 {
     color: #fff;
     background: linear-gradient(to right, #9C27B0, #E040FB);
     border: 0;
-    padding-left: 40px;
-    padding-right: 40px;
-    padding-bottom: 10px;
-    padding-top: 10px;
+    padding: 10px 40px;
```
Comment (laviealon) [inline src/main/front-end/signin/signin.css:79]: accept this

**Review by laviealon — CHANGES_REQUESTED** [review]

**Review by thelfenbaum — APPROVED** [review]

**Review by laviealon — APPROVED** [review]

**Pull request merged**